configure.py

```diff
@@ -0,0 +1,413 @@
+        ' bazel option "--config=opt" is specified [Default is %s]: '
+    ) % default_cc_opt_flags
+    cc_opt_flags = get_from_env_or_user_or_default(
+        environ_cp, "CC_OPT_FLAGS", question, default_cc_opt_flags
+    )
+    for opt in cc_opt_flags.split():
+        write_to_bazelrc("build:opt --copt=%s" % opt)
+        write_to_bazelrc("build:opt --host_copt=%s" % opt)
+    write_to_bazelrc("build:opt --define with_default_optimizations=true")
+
+
+def maybe_set_manylinux_toolchain(environ_cp):
+    write_to_bazelrc(
+        "build:manylinux2010 --crosstool_top=//third_party/toolchains/preconfig/ubuntu16.04/gcc7_manylinux2010-nvcc-cuda10.1:toolchain"
+    )
+    if get_var(
+        environ_cp,
+        var_name="MANYLINUX2010",
+        query_item="manylinux2010-compatible pip package",
+        enabled_by_default=False,
+        question=(
+            "Are you trying to build a manylinux2010-compatible pip package "
+            "in the tensorflow:custom-op-ubuntu16 Docker container?"
+        ),
+        yes_reply="Building manylinux2010-compatible pip package.",
+        no_reply="Not building manylinux2010-compatible pip package.",
+    ):
+        write_to_bazelrc("build --config=manylinux2010")
+        write_to_bazelrc("test --config=manylinux2010")
+
+
+def set_windows_build_flags(environ_cp):
+    """Set Windows specific build options."""
+
+    # First available in VS 16.4. Speeds up Windows compile times by a lot. See
+    # https://groups.google.com/a/tensorflow.org/d/topic/build/SsW98Eo7l3o/discussion
+    # pylint: disable=line-too-long
+    write_to_bazelrc(
+        "build --copt=/d2ReducedOptimizeHugeFunctions --host_copt=/d2ReducedOptimizeHugeFunctions"
+    )
+
+    if get_var(
+        environ_cp,
+        "TF_OVERRIDE_EIGEN_STRONG_INLINE",
+        "Eigen strong inline",
+        True,
+        (
+            "Would you like to override eigen strong inline for some C++ "
+            "compilation to reduce the compilation time?"
+        ),
+        "Eigen strong inline overridden.",
+        "Not overriding eigen strong inline, "
+        "some compilations could take more than 20 mins.",
+    ):
+        # Due to a known MSVC compiler issue
+        # https://github.com/tensorflow/tensorflow/issues/10521
+        # Overriding eigen strong inline speeds up the compiling of
+        # conv_grad_ops_3d.cc and conv_ops_3d.cc by 20 minutes,
+        # but this also hurts the performance. Let users decide what they want.
+        write_to_bazelrc("build --define=override_eigen_strong_inline=true")
+
+
+def reset_lce_configure_bazelrc():
+    """Reset file that contains customized config settings."""
+    open(_LCE_BAZELRC, "w").close()
+
+
+def main():
+    # Make a copy of os.environ to be clear when functions and getting and setting
+    # environment variables.
+    environ_cp = dict(os.environ)
+
+    reset_lce_configure_bazelrc()
+
+    setup_python(environ_cp)
```

Comment:
Thanks for the detailed comments on why this is a good idea. I agree that hardcoding it makes sense especially when using multiple environments to build LCE.